phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2639,6 +2646,20 @@ private MetaDataMutationResult doDropTable(byte[] key, byte[] tenantId, byte[] s
                         QueryServices.ALLOW_SPLITTABLE_SYSTEM_CATALOG_ROLLBACK);
                 }
             }
+            if (settingNewPhoenixTTLAttribute(tableMetadata, PHOENIX_TTL_BYTES)) {
```

Comment:
Circumvent this check if the tableType != VIEW to prevent unnecessary iteration of the list of mutations in non-view cases.

Reply:
In fact, since we only allow setting PHOENIX_TTL on views we should enter this `if` block only if PTableType== VIEW (maybe add it inside the `settingNewPhoenixTTLAttribute()` method itself so we can change it in the future if required).